api/src/main/java/org/pmiops/workbench/notebooks/NotebooksServiceImpl.java

```diff
@@ -127,22 +130,36 @@ public boolean isNotebookBlob(Blob blob) {
   @Override
   public FileDetail copyNotebook(
       String fromWorkspaceNamespace,
-      String fromWorkspaceName,
+      String fromWorkspaceFirecloudName,
       String fromNotebookName,
       String toWorkspaceNamespace,
-      String toWorkspaceName,
+      String toWorkspaceFirecloudName,
       String newNotebookName) {
     workspaceAuthService.enforceWorkspaceAccessLevel(
-        fromWorkspaceNamespace, fromWorkspaceName, WorkspaceAccessLevel.READER);
+        fromWorkspaceNamespace, fromWorkspaceFirecloudName, WorkspaceAccessLevel.READER);
     workspaceAuthService.enforceWorkspaceAccessLevel(
-        toWorkspaceNamespace, toWorkspaceName, WorkspaceAccessLevel.WRITER);
-    workspaceAuthService.validateActiveBilling(toWorkspaceNamespace, toWorkspaceName);
+        toWorkspaceNamespace, toWorkspaceFirecloudName, WorkspaceAccessLevel.WRITER);
+    workspaceAuthService.validateActiveBilling(toWorkspaceNamespace, toWorkspaceFirecloudName);
     newNotebookName = NotebooksService.withNotebookExtension(newNotebookName);
 
+    final DbWorkspace fromWorkspace =
+        workspaceDao.getRequired(fromWorkspaceNamespace, fromWorkspaceFirecloudName);
+    final DbAccessTier fromTier = fromWorkspace.getCdrVersion().getAccessTier();
+    final DbWorkspace toWorkspace =
+        workspaceDao.getRequired(toWorkspaceNamespace, toWorkspaceFirecloudName);
+    final DbAccessTier toTier = toWorkspace.getCdrVersion().getAccessTier();
+
+    if (!fromTier.equals(toTier)) {
+      final String msg =
+          String.format(
+              "Cannot copy from %s to %s", fromTier.getDisplayName(), toTier.getDisplayName());
```

Comment:
I think just a "Cannot copy across access tiers" might actually be more helpful since it indicates no combination of different access tiers will be valid.

Reply:
ok - I would like to keep this information however.

how about:
Cannot copy across access tiers (A to B attempted)

Reply:
sure